src/nvim/normal.c

```diff
@@ -1025,10 +1025,14 @@ static int normal_execute(VimState *state, int key)
     // If you give a count before AND after the operator, they are
     // multiplied.
     if (s->ca.count0) {
-      s->ca.count0 *= s->ca.opcount;
+      s->ca.count0 = (long)((uint64_t)s->ca.count0 * (uint64_t)s->ca.opcount);
```

Comment:
I still don't understand how (un)signed integer overflow works so I used the largest unsigned type I know and then cast the result back to long.